### ISSUE TITLE
"api_read_only" parameter is ignored in JSON config file

### ISSUE THREAD

**Issue by stralex7** [trexminer/T-Rex#901]
I have this part in my configuration:
`{
  "api_bind_http": "0.0.0.0:80",
  "api_read_only": false,
`
but the REST API is in r/o mode
version 0.24.7

**Comment by warren-ru** [comment]
Do you use Hive OS?

**Comment by trexminer** [comment]
@stralex7 you also need to set `--api-key` to allow write operations

**Comment by stralex7** [comment]
> Do you use Hive OS?
No, ubuntu 20.04

> @stralex7 you also need to set `--api-key` to allow write operations
@trexminer Anyway to override this behavior for people who understand the risk? If I bind to a particular IP instead of any will I observe the same effect?

api_bind_http is also ignored, I have to pass via command line

**Comment by trexminer** [comment]
You've misspelled the parameter names, they must have dashes, not underscores

**Comment by stralex7** [comment]
@trexminer I disagree, I took parameters from the raw config in web UI, and the rest are just working fine.

**Comment by stralex7** [comment]
I tested the binding to the particular interface/ip and the result is the same: "api_read_only": true

**Comment by trexminer** [comment]
Oh I see, you're trying to modify configuration using API. In that case you need to pass in your session id, like shown here: https://github.com/trexminer/T-Rex/wiki/API (see **login** handler).

**Comment by stralex7** [comment]
Thank you, but I need to keep unauthenticated access to /summary, for my Prometheus collector.
